clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TaskHealthCachingAgent.java

```diff
@@ -129,34 +129,38 @@ public TaskHealthCachingAgent(
             Keys.getTaskDefinitionKey(accountName, region, service.getTaskDefinition());
         TaskDefinition taskDefinition = taskDefinitionCacheClient.get(taskDefinitionCacheKey);
 
-        if (isContainerMissingNetworking(task)) {
+        if (taskDefinition == null) {
           log.debug(
-              "Task '{}' is missing networking. Will not retrieve health.", task.getTaskArn());
+              "Provided task definition '{}' is null for service '{}'",
+              service.getTaskDefinition(),
+              service.getServiceName());
+          continue;
+        }
+
+        boolean lacksNetworkBindings = isTaskMissingNetworkBindings(task);
+        if (task.getContainers().isEmpty()
+            || (lacksNetworkBindings && isTaskMissingNetworkInterfaces(task))) {
```

Comment:
nit: we should probably also move isTaskMissingNetworkInterfaces to a variable (seems weird that lacksNetworkBindings is a variable while lacksNetworkInterfaces is not or vice versa)

Reply:
Also, technically a task should not be able to have both networkingBindings and networkInterfaces (can we add a check that logs an error if this were ever to happen - just so we're made aware of any weird behaviour) 

`          if (!lacksNetworkBindings && !lacksNetworkInterfaces) {`

Reply:
originally i had both pulled out as vars, but the result of `isTaskMisingNetworkingInterfaces` was never used again, so I decided to just call the func here. Can put it back if there's a benefit to variable-izing it (not sure that's a word...)

Reply:
I think it makes sense if it's used here as well https://github.com/spinnaker/clouddriver/pull/4712#discussion_r445765662